src/pulselistener/pulselistener/lib/bus.py

```diff
@@ -99,7 +100,15 @@ def add_queue(self, name, mp=False, redis=False, maxsize=-1):
 
         else:
             # Run the synchronous mp queue.get in the asynchronous loop
-            return await asyncio.get_running_loop().run_in_executor(None, queue.get)
+            # but use an asyncio sleep to be able to react to cancellation
```

Comment:
I wonder if there isn't a nicer way to achieve this

Reply:
I can rollback to the initial code here, as it is replaced by Redis queue on Heroku (and even dev)